scripts.py

```diff
@@ -38,6 +55,310 @@ def run(self):
         VIAFClient(self._db).run(self.force)
 
 
+
+class CheckContributorTitles(Script):
+    """ TODO """
+
+    def __init__(self, viaf=None):
+        self.viaf = viaf or VIAFClient(self._db)
+
+
+    def run(self, batch_size=1000):        
+        query = self._db.query(Contributor).filter(Contributor.viaf!=None).order_by(Contributor.id)
+
+        if self.log:
+            self.log.info(
+                "Processing %d contributors.", query.count()
+            )
+
+        contributors = True
+        offset = 0
+        output = "ContributorID|\tSortName|\tTitle"
+        print output.encode("utf8")
+        from core.model import dump_query
+        while contributors:
+            my_query = query.offset(offset).limit(batch_size)
+
+            #print "query=%s" % dump_query(my_query)
+            contributors = my_query.all()
+
+            for contributor in contributors:
+                self.process_contributor(contributor)
+            offset += batch_size
+            #self._db.commit()
+        #self._db.commit()
+
+
+    def process_contributor(self, contributor):
+        if not contributor or not contributor.viaf:
+            return
+
+        # we should have enough known viaf ids for our task to only process those 
+        contributor_titles = self.viaf.lookup_name_title(contributor.viaf)
+        if contributor_titles:
+            output = "%s|\t%s|\t%r" % (contributor.id, contributor.sort_name, contributor_titles)
+            print output.encode("utf8")
+
+
+
+class CheckContributorNamesOnWeb(CheckContributorNamesInDB):
+    """
+    Inherits process_contribution_local from parent.  
+    Adds process_contribution_viaf functionality, which 
+    sends a request to viaf to try and determine correct sort_name 
+    for a given author.  
+    """
+
+    COMPLAINT_SOURCE = "CheckContributorNamesOnWeb"
+
+
+    def __init__(self, _db=None, cmd_args=None):
+        super(CheckContributorNamesOnWeb, self).__init__(_db=_db)
+
+        parsed_args = self.parse_command_line(_db=self._db, cmd_args=cmd_args)
+        self.mock_mode = parsed_args.mock
+
+        if self.mock_mode:
+            self.log.debug(
+                "This is mocked run, with metadata coming from test files, rather than live OneClick connection."
+            )
+            self.base_path = os.path.split(__file__)[0]
+            self.base_path = os.path.join(self.base_path, "tests")
+            self.canonicalizer = MockAuthorNameCanonicalizer(self._db)
```

Comment:
I mentioned changing the way you mock scripts and this is what I'm talking about. Rather than passing in a `mock_mode` variable I recommend passing in the AuthorNameCanonicalizer class you want to use.